lib/history.rb

```diff
@@ -20,38 +20,33 @@ def write_history(image_name)
     end
   end
 
-  def should_adopt?(image_name, filename_list)
+  def should_adopt?(image_name, dropbox_files)
     create_history_file_if_needed
 
-    json_data = open(HISTORY_JSON_FILE_PATH) do |io|
+    counts_by_filename = open(HISTORY_JSON_FILE_PATH) do |io|
       JSON.load(io)
     end
 
-    # NOTE: nil.to_i => 0
-    count = json_data[image_name].to_i
-    min_count = calculate_min_count(json_data, filename_list)
+    count = counts_by_filename[image_name] || 0
+    min_count = calculate_min_count(counts_by_filename, dropbox_files)
 
     count <= min_count
   end
 
-  # NOTE: To alculate only once, use instance variable
-  def calculate_min_count(json_data, filename_list)
-    return @min_count unless @min_count.nil?
```

Comment:
As Note says, `should_adopt?` will be called every time, if the candidate are not the file selected the least number of times. So in the worst case, the number of trials is the number of files in dropbox.

IMO: To avoid such case, we should use an instance variable which can hold values ​​permanently.

Reply:
Hmm... Considering the method alone, I think it is wrong to memorize the return value because the arguments are not fixed values. 

If you want to save the return value, you need to fix the input (for example, change to a class and receive input in the constructor). However, I think it is unnecessary because it becomes complicated.

Reply:
> it is wrong to memorize the return value because the arguments are not fixed values

I see. I agree with this idea 
As you say, this kind of tool's source code should be simple and **too much** optimization will raise some tiny bugs.

Thx for re-considering 👍